Source/messaging/MessageUnit.cpp

```diff
@@ -145,18 +145,29 @@ namespace WPEFramework {
         {
             uint32_t result = Core::ERROR_OPENING_FAILED;
 
-            string basePath = Core::Directory::Normalize(pathName) + _T("MessageDispatcher");
+            uint32_t permission = 0;
```

Comment:
Code for getting the rights is already here: https://github.com/rdkcentral/Thunder/blob/3249f7cae8588a8d7022f03edfc9d9c0e412ed43/Source/core/NodeId.cpp#L373

Let's not have this mutliple times. I prose moving the rights code from NodeId.cpp to somewhere else (I think FileSystem might be a good place). Then here as well in NodeId we use the same code to extract the access rights)

Reply:
moved to FileSytem.h